scope/test.h

```diff
@@ -52,14 +52,18 @@ namespace scope {
   }
 
   // eval_equal_impl for anything having std::begin and std::end overloads
-  template <typename ExceptionType, typename ExpSequenceT, typename ActSequenceT>
+  template <
+    typename ExceptionType,
+    typename ExpSequenceT,
+    typename ActSequenceT
+  >
   auto eval_equal_impl(ExpSequenceT&& e, ActSequenceT&& a, int, const char* const file, int line, const char* msg = "") -> decltype(std::begin(e), std::end(e), std::begin(a), std::end(a), void()) {
     const auto abeg = std::begin(a);
     const auto aend = std::end(a);
     const auto ebeg = std::begin(e);
     const auto eend = std::end(e);
-    const size_t elen = eend - ebeg;
-    const size_t alen = aend - abeg;
```

Comment:
Since `std::distance` may not be O(1) for all container types, I think what would be better is to call `std::mismatch` to perform the comparison. If there is a mismatch, _then_ the size of each container/range should be computed, inside the `if (mis.first != eend)` block.